Uchu.World/Objects/Components/Player/InventoryManagerComponent.cs

```diff
@@ -140,7 +146,9 @@ public async Task SaveAsync(UchuContext context)
             // Update existing items and add new items
             foreach (var itemToSave in itemsToSave)
             {
-                var savedItem = character.Items.FirstOrDefault(i => i.Id == itemToSave.Id); 
+                var savedItem = character.Items.FirstOrDefault(i => i.Id == itemToSave.Id);
+                var inventoryType = itemToSave.Inventory?.InventoryType ?? InventoryType.None;
+                if (inventoryType == InventoryType.VendorBuyback) continue;
```

Comment:
`itemToSave` directly points to `Items` with no modifications. Since you are filtering sold items out, it looks like you can just use the `UnsoldItems` property you added.

Reply:
That change makes sense to me. I will need to reach out to my professor to see if I can consider this done or not.